packages/react/src/components/PageHeader/PageHeader.tsx

```diff
@@ -241,6 +235,186 @@ PageHeaderContent.propTypes = {
+  children?: React.ReactNode;
+  /**
+   * Specify an optional className to be added to your PageHeaderContentPageActions
+   */
+  className?: string;
+  /**
+   *  The PageHeaderContent's page actions collabpsible Menu button label
+   */
+  menuButtonLabel?: string;
+  /**
+   * The PageHeaderContent's page actions
+   */
+  pageActions?: React.ReactNode;
+}
+const PageHeaderContentPageActions = React.forwardRef<
+  HTMLDivElement,
+  PageHeaderContentPageActionsProps
+>(function PageHeaderContentPageActions(
+  {
+    className,
+    children,
+    menuButtonLabel = 'Actions',
+    pageActions,
+    ...other
+  }: PageHeaderContentPageActionsProps,
+  ref
+) {
+  const prefix = usePrefix();
+  const classNames = classnames(
+    {
+      [`${prefix}--page-header__content__page-actions`]: true,
+    },
+    className
+  );
+  const containerRef = useRef<HTMLDivElement>(null);
+  const offsetRef = useRef<HTMLDivElement>(null);
+  const [menuButtonVisibility, setMenuButtonVisibility] = useState(false);
+  const [hiddenItems, setHiddenItems] = useState<any[]>([]);
+
+  // need to set the grid columns width based on the menu button's width
+  // to avoid overlapping when resizing
+  useLayoutEffect(() => {
+    if (menuButtonVisibility && offsetRef.current) {
+      console.log('hello');
+      const width = offsetRef.current.offsetWidth;
+      document.documentElement.style.setProperty(
+        '--pageheader-title-grid-width',
+        `${width}px`
+      );
+    }
+  }, [menuButtonVisibility]);
+
+  useEffect(() => {
+    if (!containerRef.current || !Array.isArray(pageActions)) return;
+    createOverflowHandler({
+      container: containerRef.current,
+      // exclude the hidden menu button from children
+      maxVisibleItems: containerRef.current.children.length - 1,
+      onChange: (visible, hidden) => {
+        setHiddenItems(pageActions?.slice(visible.length));
+
+        if (hidden.length > 0) {
+          setMenuButtonVisibility(true);
+        }
+      },
+    });
+  }, []);
+
+  return (
+    <div className={classNames} ref={containerRef}>
+      {pageActions && (
+        <>
+          {Array.isArray(pageActions) && (
+            <>
+              {pageActions.map((action) => (
+                <div key={action.id} className="action">
+                  {action.body}
+                </div>
+              ))}
+              <span data-offset data-hidden ref={offsetRef}>
+                <MenuButton
+                  menuAlignment="bottom-end"
+                  label={menuButtonLabel}
+                  size="md">
+                  {hiddenItems &&
+                    hiddenItems.length > 0 &&
+                    hiddenItems
+                      .reverse()
+                      .map((item) => (
+                        <MenuItem
+                          key={item.id}
+                          label={item.label}
+                          onClick={item.onClick}
+                        />
+                      ))}
```

Comment:
If the types are correct and `hiddenItems` is always an array, I don't think the conditions here should be necessary. It seems like you could simplify quite a bit.

```suggestion
                  {hiddenItems.reverse().map((item) => (
                    <MenuItem
                      key={item.id}
                      label={item.label}
                      onClick={item.onClick}
                    />
                  ))}
```